### DISCUSSION TITLE
db의 카테고리와 관련된 칼럼은 어떻게 설계하면 좋을까요

### DISCUSSION THREAD

**Discussion by lky3004me** [100-hours-a-week/discussions#66] [개발] [answered]
현재 다나와에서 닭가슴살 제품의 카테고리를 긁어오면 다음과 같습니다.

닭고기, 닭가슴살, 그릴·스팀
닭고기, 닭가슴살
닭고기, 닭가슴살, 스테이크

현재 저희는 카테고리라는 칼럼 하나에 카테고리 관련된 모든 정보를 저장하는 상황입니다.

위와 같은 방식의 문제점은 다음과 같습니다.
1. 검색관련 기능에서, "닭가슴살"이라고 검색을 하면 순수 닭가슴살만 검색되지 않고, "닭가슴살, 소시지"와 같이 포함되서 검색됩니다.
2. 프론트측에서 카테고리를 보여줄 때 항상 뭉쳐서 오기 때문에 나눠달라는 요청을 받았었습니다.

위와 관련된 사항때문에 카테고리를 분리하고싶습니다만, 여러가지 고민이 생깁니다.

그러므로 보통 카테고리를 설계할 때 어떤 방식으로 하는지 궁금합니다. 또한 현재 저희의 문제사항에 적절한 방법은 대분류와 소분류의 칼럼 2개를 만들어서, 소분류에서 검색이 되게 만드는 것인데 이러한 방법에 대해 어떻게 생각하시는지 궁금합니다

**Comment by ej31** [comment, marked as answer]
# 답변

- 제가 잘 이해한게 맞는지 모르겠지만 ..

- 카테고리는 특정 객체를 묶어주는 역할을 합니다.
- 다나와의 카테고리가 무엇이든 상관없이 결국 내 서비스의 비즈니스 로직에 맞게 카테고리를 설계하면 될 일이지요
  - 굳이 다나와를 의식해서 카테고리를 설계할 필요 없다는 뜻입니다.

- 만약 저라면 그냥 상품 이름 안에 닭가슴살이 포함되어 있으면 내려줄 것 같습니다.
- 아니면 상품을 트리구조로 세분화해서 대중소분류로 나눠서 카테고라이징 할 수도 있겠지요

- 이건 서비스 구조하고 DB 설계 어떻게 하셨는지 봐야 답변 드릴 수 있을 것 같습니다. 내일 자리로 갈께요!
